contracts/base/escrow/dapps/Curve.sol

```diff
@@ -0,0 +1,267 @@
+pragma solidity 0.5.17;
+pragma experimental ABIEncoderV2;
+
+// External Libraries
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/ERC20Detailed.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/utils/Address.sol";
+
+//Contracts
+import "../../BaseEscrowDapp.sol";
+
+// Interfaces
+import "./ICurve.sol";
+import "../../../providers/curve/ICurvePool.sol";
+import "../../../providers/curve/ICurveGauge.sol";
+import "../../../providers/curve/ICurveMinter.sol";
+import "../../../providers/curve/ICurveVoting.sol";
+import "../../../providers/curve/ICurveFeeDistribution.sol";
+
+/*****************************************************************************************************/
+/**                                             WARNING                                             **/
+/**                      DAPP CONTRACT IS AN EXTENSION OF THE ESCROW CONTRACT                       **/
+/**  ---------------------------------------------------------------------------------------------  **/
+/**  Because there are multiple dApp contracts, and they all extend the Escrow contract that is     **/
+/**  itself upgradeable, they cannot have their own storage variables as they would cause the the   **/
+/**  storage slots to be overwritten on the Escrow proxy contract!                                  **/
+/**                                                                                                 **/
+/**  Visit https://docs.openzeppelin.com/upgrades/2.6/proxies#upgrading-via-the-proxy-pattern for   **/
+/**  more information.                                                                              **/
+/*****************************************************************************************************/
+/**
+    @notice This contract is used to define Curve dApp actions available. All dapp actions are invoked via
+        delegatecalls from Escrow contract, so this contract's state is really Escrow.
+    @author develop@teller.finance
+ */
+contract Curve is ICurve, BaseEscrowDapp {
+    using Address for address;
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    /** External functions */
+    /**
+        @notice Deposits tokens into a Curve pool, and then stakes the LP tokens received in a Curve Gauge contract
+        @param tokenAddress address of the asset token contract to deposit
+        @param amount amount of tokens to deposit
+    */
+    function deposit(address tokenAddress, uint256 amount) public onlyOwner() {
```

Comment:
I would add as a reference the link to the Curve pool. Which would it be?

https://github.com/curvefi/curve-contract/tree/master/contracts/pools/3pool
https://github.com/curvefi/curve-contract/tree/master/contracts/pools/y

Pools: https://github.com/curvefi/curve-contract/tree/master/contracts/pools